README.md

```diff
@@ -1,2 +1,120 @@
-# mailgun-relayery
-A Go server providing an additional security layer for using the MailGun API
+Mailgun Relayery
+================
+
+Mailgun Relayery is a tool providing an additional security layer for the MailGun API.
+
+**The problem**: MailGun gives out a single API key to each account. For entities dealing with multiple clients or 
+applications, giving out their API key to each of them constitutes a blatant security violation.
+
+**The solution**: _Mailgun Relayery_ acts as a middle layer between message senders and the MailGun API.
+Senders are issued descriptor strings (acting like usernames) and HTTP authentication tokens, which 
+are used to authenticate their requests to the middle layer and to relay their messages to MailGun.
+
+The tool stores authentication data (HTTP tokens) and channeling data (mailing fields like recipients, cc, bcc, etc) 
+in an LMDB, guaranteeing persistency. Two Go servers have access to the LMDB:
+
+* the Control server (with read-write access to the LMDB) manages authentication data and mailing channels; this 
+    server should only be accessed via secure connection, e.g. ssh, and managed by a trusted party.
+* the Relay server (with read-only access to the LMDB) receives and authenticates HTTP requests to relay 
+    messages to the MailGun API; this server is open to the whole internet.
```

Comment:
s/internet/Internet? Please double-check, I'm not positive.

Reply:
Internet is the [most popular](https://blog.oxforddictionaries.com/2016/04/05/should-you-capitalize-internet/).